### PR TITLE
added controllers listed in task

### PR DESCRIPTION
**Pull request by RomanDubovskyi**
added controllers listed in task
Dto created
improved Service and Dao layers in order for Controllers to function properly

### CODE REVIEW

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class AuthenticationController {
+    private static final Logger LOGGER = Logger.getLogger(AuthenticationController.class);
+    private final AuthenticationService authenticationService;
+
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/register")
+    public void registerUser(@RequestBody UserRequestDto userRequestDto) {
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: void краще не повертати

File: src/main/java/com/dev/cinema/controllers/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private final ShoppingCartService shoppingCartService;
+
+    public OrderController(OrderService orderService,
+                           UserService userService, ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @PostMapping(value = "/complete-order")
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/OrderController.java:36]:
```suggestion
    @PostMapping(value = "/complete")
```

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    @PostMapping(value = "add-movie-session")
+    public void addMovieSessionToCart(@RequestBody MovieSessionRequestDto movieSessionRequestDto,
+                                      @RequestParam Long userId) {
+        MovieSession movieSession = transferDtoToMovieSession(movieSessionRequestDto);
+        User user = userService.getById(userId);
+        shoppingCartService.addSession(movieSession, user);
+    }
+
+    @GetMapping(value = "by-user")
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]:
```suggestion
    @GetMapping(value = "/get/user")
```

File: src/main/java/com/dev/cinema/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    @PostMapping(value = "/add")
+    public void addUser(@RequestBody UserRequestDto userRequestDto) {
+        User newUser = new User();
+        newUser.setEmail(userRequestDto.getEmail());
+        newUser.setPassword(userRequestDto.getPassword());
+        userService.add(newUser);
+    }
+
+    @GetMapping(value = "/by-email")
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/UserController.java:32]:
```suggestion
    @GetMapping(value = "/get/email")
```

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("shopping-carts")
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]:
```suggestion
@RequestMapping("/shopping-carts")
```

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  MovieService movieService, CinemaHallService cinemaHallService,
+                                  UserService userService) {
+        this.shoppingCartService = shoppingCartService;
+        this.cinemaHallService = cinemaHallService;
+        this.movieService = movieService;
+        this.userService = userService;
+    }
+
+    @PostMapping(value = "add-movie-session")
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]:
```suggestion
    @PostMapping(value = "add-movie-session")
```
```suggestion
    @PostMapping(value = "/add-movie-session")
```

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 7]
```diff
@@ -59,4 +59,9 @@ public MovieSession add(MovieSession movieSession) {
             throw new RuntimeException("Can't insert MovieSession entity", e);
         }
     }
+
+    @Override
+    public MovieSession getById(Long id) {
+        return null;
```
Comment (Tarabarnik) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: why null?

Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: didn't use it, forgot to delete id, sry

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: changed it everywhere, now it returns object that was created

**Review by Tarabarnik — APPROVED** [review]

**Pull request merged**